worlds/mlss/Client.py

```diff
@@ -0,0 +1,266 @@
+from worlds._bizhawk.client import BizHawkClient
+
+if TYPE_CHECKING:
+    from worlds._bizhawk.context import BizHawkClientContext
+else:
+    BizHawkClientContext = object
+
+ROOM_ARRAY_POINTER = 0x51fa00
+
+# Add .apmlss suffix to bizhawk client
+from worlds.LauncherComponents import SuffixIdentifier, components
+for component in components:
+    if component.script_name == "BizHawkClient":
+        component.file_identifier = SuffixIdentifier(*(*component.file_identifier.suffixes, ".apmlss"))
+        break
+
+
+class MLSSClient(BizHawkClient):
+    game = "Mario & Luigi Superstar Saga"
+    system = "GBA"
+    local_checked_locations: Set[int]
+    goal_flag: int
+    rom_slot_name: Optional[str]
+    player: int
+    eCount: int
+    eUsed: [int]
+    player_name: Optional[str]
+    checked_flags: dict[int, list] = {}
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.local_checked_locations = set()
+        self.local_set_events = {}
+        self.local_found_key_items = {}
+        self.rom_slot_name = None
+        self.lock = asyncio.Lock
+        self.eCount = 0
+        self.eUsed = []
+
+    async def validate_rom(self, ctx: BizHawkClientContext) -> bool:
+        from CommonClient import logger
+
+        try:
+            # Check ROM name/patch version
+            rom_name_bytes = ((await bizhawk.read(ctx.bizhawk_ctx, [(0xA0, 14, "ROM")]))[0])
+            rom_name = bytes([byte for byte in rom_name_bytes if byte != 0]).decode("UTF-8")
+            if not rom_name.startswith("MARIO&LUIGIU"):
+                return False
+            if rom_name == "MARIO&LUIGIUA8":
+                logger.info("ERROR: You appear to be running an unpatched version of Mario & Luigi Superstar Saga. "
+                            "You need to generate a patch file and use it to create a patched ROM.")
+                return False
+            if rom_name != "MARIO&LUIGIUAP":
+                logger.info("ERROR: The patch file used to create this ROM is not compatible with "
+                            "this client. Double check your client version against the version being "
+                            "used by the generator.")
+                return False
+        except UnicodeDecodeError:
+            return False
+        except bizhawk.RequestFailedError:
+            return False  # Should verify on the next pass
+
+        ctx.game = self.game
+        ctx.items_handling = 0b101
+        ctx.want_slot_data = True
+        ctx.watcher_timeout = 0.125
+        self.rom_slot_name = rom_name
+        name_bytes = ((await bizhawk.read(ctx.bizhawk_ctx, [(0xB0, 16, "ROM")]))[0])
+        player = ((await bizhawk.read(ctx.bizhawk_ctx, [(0xAF, 1, "ROM")]))[0][0])
+        self.player = player + 1
+        name = bytes([byte for byte in name_bytes if byte != 0]).decode("UTF-8")
+        self.player_name = name
+        logger.info(name)
+
+        for i in range(59):
+            self.checked_flags[i] = [9, 10]
```

Comment:
What's the purpose of adding 9 and 10? The elements are meant to represent bits in a byte, right? Which would put 9 and 10 out of bounds anyway.